### PR TITLE
Async Main on templates

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes https://github.com/aspnet/Templating/issues/733.

### CODE REVIEW

**Comment by davidfowl** [comment]
@ryanbrandenburg do you know if this works in VS 16 OOTB? I was playing around with it and it seems like async main isn't on by default. 

@jaredpar it seems like the default langauge version of 16 isn't quite 8? When will we be able to use async main by default without a language version specified in the project template?

File: src/Templating/src/Microsoft.DotNet.Web.ProjectTemplates/content/EmptyWeb-CSharp/Program.cs [hunk 1]
```diff
@@ -12,9 +12,9 @@ namespace Company.WebApplication1
 {
     public class Program
     {
-        public static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
-            CreateWebHostBuilder(args).Build().Run();
+            await CreateWebHostBuilder(args).Build().RunAsync();
```
Comment (khellang) [inline src/Templating/src/Microsoft.DotNet.Web.ProjectTemplates/content/EmptyWeb-CSharp/Program.cs]: Y NO `return`? :thinking:

Comment (benaadams) [reply src/Templating/src/Microsoft.DotNet.Web.ProjectTemplates/content/EmptyWeb-CSharp/Program.cs]: For templates establish the pattern before the optimization?

Straight return `Task rather` than `await Task` has caveats; like don't do it in a `using`, try/catch/finally etc

Comment (davidfowl) [reply src/Templating/src/Microsoft.DotNet.Web.ProjectTemplates/content/EmptyWeb-CSharp/Program.cs]: Yes, awaiting is fine here.

**Comment by jaredpar** [comment]
> it seems like the default langauge version of 16 isn't quite 8? 

Correct. The C# 8 features are in preview at Dev16 RTM. They won't go RTM themselves until we get to .NET Core 3.0 RTM.

> When will we be able to use async main by default without a language version specified in the project template?

Likely Dev16RTM when we move the default language version to 7.3.

**Comment by mkArtakMSFT** [comment]
@ryanbrandenburg we should probably close this and reopen when we will get to this.

**Pull request closed without merging**